packages/dx-scheduler-core/src/plugins/drag-drop-provider/helpers.ts

```diff
@@ -63,46 +65,84 @@ export const autoScroll: PureComputed<
   }
 };
 
-export const calculateAppointmentTimeBoundaries: CalculateAppointmentTimeBoundaries = (
+export const timeBoundariesByResize: TimeBoundariesByResize = (
+  payload, targetData, targetType, cellDurationMinutes, insidePart,
+) => {
+  if (targetType !== payload.appointmentType) {
+    return { appointmentStartTime: undefined, appointmentEndTime: undefined };
+  }
+
+  let appointmentStartTime;
+  let appointmentEndTime;
+  const sourceType = payload.type;
+
+  if (sourceType === RESIZE_TOP) {
+    const insideTopOffset = targetType === VERTICAL_TYPE
+      ? insidePart * cellDurationMinutes * 60 / 2 : 0;
+    appointmentStartTime = moment(targetData.startDate as Date)
+      .add(insideTopOffset, SECONDS).toDate();
+    appointmentEndTime = new Date(payload.endDate as Date);
+  }
+  if (sourceType === RESIZE_BOTTOM) {
+    const insideBottomOffset = insidePart === 0 && targetType === VERTICAL_TYPE
+      ? cellDurationMinutes * 60 / 2 : 0;
+    appointmentEndTime = moment(targetData.endDate as Date)
+      .add(-insideBottomOffset, SECONDS).toDate();
+    appointmentStartTime = new Date(payload.startDate as Date);
+  }
+  // keep origin appointment duration if coordinates are wrong
+  if (moment(appointmentEndTime).diff(appointmentStartTime, MINUTES) < 1) {
+    appointmentStartTime = new Date(payload.startDate as Date);
+    appointmentEndTime = new Date(payload.endDate as Date);
+  }
+  return { appointmentStartTime, appointmentEndTime };
+};
+
+export const timeBoundariesByDrag: TimeBoundariesByDrag = (
   payload, targetData, targetType,
   cellDurationMinutes, insidePart, offsetTimeTopBase,
 ) => {
-  const appointmentDurationSeconds = intervalDuration(payload, SECONDS);
-  const sourceType = payload.type;
-  const insideOffset = targetType === VERTICAL_TYPE ? insidePart * cellDurationMinutes * 60 / 2 : 0;
   let offsetTimeTop;
+  let appointmentStartTime;
+  let appointmentEndTime;
+
+  const insideOffset = targetType === VERTICAL_TYPE ? insidePart * cellDurationMinutes * 60 / 2 : 0;
+  const start = moment(targetData.startDate as Date).add(insideOffset, SECONDS);
+  const end = moment(start);
```

Comment:
Looks like the `end` value isn't required here. It's only used here - `appointmentEndTime = moment(end)`. Can be changed to `appointmentEndTime = moment(start)`.